homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -91,17 +135,168 @@ async def async_setup_entry(hass, config_entry):
     return True
 
 
+async def async_setup_lip(hass, config_entry, lip_devices):
+    """Connect to the bridge via Lutron Integration Protocol to watch for pico remotes."""
+    host = config_entry.data[CONF_HOST]
+    config_entry_id = config_entry.entry_id
+    data = hass.data[DOMAIN][config_entry_id]
+    bridge_device = data[BRIDGE_DEVICE]
+    bridge = data[BRIDGE_LEAP]
+    lip = LIP()
+    try:
+        await lip.async_connect(host)
+    except asyncio.TimeoutError:
+        _LOGGER.error("Failed to connect to via LIP at %s:23", host)
+        return
+
+    _LOGGER.debug("Connected to Lutron Caseta bridge via LIP at %s:23", host)
+    button_devices_by_lip_id = _async_merge_lip_leap_data(lip_devices, bridge)
+    button_devices_by_dr_id = await _async_register_button_devices(
+        hass, config_entry_id, bridge_device, button_devices_by_lip_id
+    )
+    _async_subscribe_pico_remote_events(hass, lip, button_devices_by_lip_id)
```

Comment:
~~Do you need to `await` this?~~

Saw below the method is not async.

Reply:
Its decorated with `@callback` to indicate it is safe to run in async context (and if we ever pass it to hass.async_add_job it will run in async).